File: python/DESEncryption.py
# older encryption standard
# data encryption standard 
# no longer considered secure for most modern applications due to its relatively smalll key size 

from Crypto.Cipher import DES
from Crypto.Util.Padding import pad, unpad 

def encrypt_des(message, key):
  des = DES.new(key, DES.MODE_ECB)
  padded_message = pad(message.encode(), DES.block_size)
  encrypted_message = des.encrypt(padded_message)
  return encrypted_message

def decrypt_des(encrypted_message, key):
  des = DES.new(key, DES.MODE_ECB)
  decrypted_message = des.decrypt(encrypted_message)
  return unpad(decrypted_message, DES.block_size).decode()


# example usage
key = b'secretkey' 
message = "hello, world!"

encrypted_message = encrypt_des(message, key)
print("encrypted message: ", encrypted_message)

decrypted_message = decrypt_des(encrypted_message, key)
print("decrypted message:", decrypted_message)

File: java/DESEncryption.java
import javax.crypto.Cipher;
import javax.crypto.SecretKeyFactory;
import javax.crypto.spec.DESKeySpec;

public class DESEncryption
  {
    public static byte[] encrypt(String plainText, String key) throws Exception {
      Cipher cipher = Cipher.getInstance("DES/ECB/PKCS5Padding");
      SecretKeyFactory factory = SecretKeyFactory.getInstance("DES");
      DESKeySpec dks = new DESKeySpec(key.getBytes());
      javax.crypto.SecretKey secretKey = factory.generateSecret(dks);
      cipher.init(Cipher.ENCRYPT_MODE, secretKey);
      byte[] encryptedText = cipher.doFinal(plainText.getBytes());
      return encryptedText;
    }

    public static String decrypt(byte[] encryptedText, String key) throws Exception
    {
      Cipher cipher = Cipher.getInstance("DES/ECB/PKCS5Padding");
      SecretKeyFactory factory = SecretKeyFactory.getInstance("DES");
      DESKeySpec dks = new DESKeySpec(key.getBytes());
      javax.crypto.SecretKey secretKey = factory.generateSecret(dks);
      cipher.init(Cipher.DECRYPT_MODE, secretKey);
      byte[] decryptedText = cipher.doFinal(encryptedText);
      return new String(decryptedText);
    }
    public static void main(String[] args) throws Exception
    {
      String secretKey = "yourSecretKey";
      String plainText = "Hello, world!";

      byte[] encryptedText = encrypt(plainText, secretKey);
      System.out.println("encrypted text: " + new String(encryptedText));

      String decryptedText = decrypt(encryptedText, secretKey);
      System.out.println("Decrypted text: " + decryptedText);
    }
  }
